### PR TITLE
refactor: change error message format to multiline

### PR DESCRIPTION
**Pull request by callmehiphop**
Fixes https://github.com/googleapis/nodejs-bigquery/issues/326
- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

---

Using the following server error for the before/after

```json
{
  "state": "DONE",
  "errorResult": {
    "reason": "invalid",
    "location": "",
    "message": "Error while reading data, error message: JSON table encountered too many errors, giving up. Rows: 1; errors: 1. Please look into the errors[] collection for more details."
  },
  "errors": [
    {
      "reason": "invalid",
      "location": "",
      "message": "Error while reading data, error message: JSON table encountered too many errors, giving up. Rows: 1; errors: 1. Please look into the errors[] collection for more details."
    },
    {
      "reason": "invalid",
      "location": "",
      "message": "Error while reading data, error message: JSON parsing error in row starting at position 0: Could not convert value to string. Field: caller; Value: 822"
    }
  ]
}
```

Before

```
Error while reading data, error message: JSON table encountered too many errors, giving up. Rows: 1; errors: 1. Please look into the errors[] collection for more details.
    at endReadableNT (_stream_readable.js:1081:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

After

```
Multiple errors occurred during the request. Please see the `errors` array for complete details.

    1. Error while reading data, error message: JSON table encountered too many errors, giving up. Rows: 1; errors: 1. Please look into the errors[] collection for more details.
    2. Error while reading data, error message: JSON parsing error in row starting at position 0: Could not convert value to string. Field: caller; Value: 822

    at endReadableNT (_stream_readable.js:1081:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

### CODE REVIEW

**Comment by callmehiphop** [comment]
@JustinBeckwith I think if users did any kinds of string/regexp operations against the error message, it would definitely throw them for a loop. If preferable, I could just make some new custom errors in the BigQuery lib itself, but it would still be a breaking change in the same capacity, which would require a semver major bump.

**Comment by stephenplusplus** [comment]
Can you do a before / after?

**Comment by stephenplusplus** [comment]
I didn't see it. Could you do one for here anyway in the opening post so it's easier for users and future us? :)

**Comment by callmehiphop** [comment]
@stephenplusplus I just updated the overview with some examples. PTAL!

**Comment by stephenplusplus** [comment]
This does make it easier to read, but since we're here, we could probably kick it up a notch:

```
Multiple errors occurred during the request. Please see the `errors` array for complete details.

    1. Error while reading data, error message: JSON table encountered too many errors, giving up. Rows: 1; errors: 1. Please look into the errors[] collection for more details.
    2. Error while reading data, error message: JSON parsing error in row starting at position 0: Could not convert value to string. Field: caller; Value: 822

    at endReadableNT (_stream_readable.js:1081:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

**Comment by callmehiphop** [comment]
@stephenplusplus SGTM - PTAL :)

File: src/util.ts [hunk 1]
```diff
@@ -199,21 +199,44 @@ export class ApiError extends Error {
...
-    if (this.errors && this.errors.length === 1) {
-      messages.push(this.errors[0].message!);
-    } else if (this.response && this.response.body) {
-      messages.push(ent.decode(errorBody.response.body.toString()));
-    } else if (!errorBody.message) {
-      messages.push('Error during request.');
+    let messageArr: string[] = Array.from(messages);
+
+    if (messageArr.length > 1) {
+      messageArr = messageArr.map((message, i) => `    ${i + 1}. ${message}`);
```
Comment (stephenplusplus) [inline src/util.ts]: I think the `\t` here is the way to go.

Comment (callmehiphop) [reply src/util.ts]: I'm not sure that is correct, on my local machine tabs get formatted like so

```js
const e = new Error('hi\n\thi');
console.error(e)
```

```sh
Error: hi
	hi
    at repl:1:11
    at Script.runInThisContext (vm.js:96:20)
    at REPLServer.defaultEval (repl.js:329:29)
    at bound (domain.js:396:14)
    at REPLServer.runBound [as eval] (domain.js:409:12)
    at REPLServer.onLine (repl.js:627:10)
    at REPLServer.emit (events.js:187:15)
    at REPLServer.EventEmitter.emit (domain.js:442:20)
    at REPLServer.Interface._onLine (readline.js:290:10)
    at REPLServer.Interface._line (readline.js:638:8)
```

Maybe we're better off avoiding indenting altogether

Comment (JustinBeckwith) [reply src/util.ts]: I love it just the way it is ❤️

Comment (stephenplusplus) [reply src/util.ts]: Yeah, the look of the spaces is my preference as well. I thought using tabs was some type of Unix standard, but I haven't found anything to support that. So, let's just use spaces!

**Comment by stephenplusplus** [comment]
@callmehiphop would you mind updating the first post with the new formatting?

**Comment by callmehiphop** [comment]
@stephenplusplus sure, done!

**Comment by oliverwoodings** [comment]
Thanks so much for getting this done! Love how quickly you guys pick up and resolve issues.

**Pull request merged**